src/IronRure/CapturesIter.cs

```diff
@@ -2,51 +2,64 @@
 using System.Collections.Generic;
 
-namespace IronRure
+namespace IronRure;
+
+/// <summary>
+///     An enumerator of regex captures. Uses the underlying Rure
+///     regex iterator.
+/// </summary>
+/// <remarks>Initialise a captures iterator.</remarks>
+public class CapturesIter : RegexIter, IEnumerator<Captures>
 {
     /// <summary>
-    ///   An enumerator of regex captures. Uses the underlying Rure
-    ///   regex iterator.
+    ///     An enumerator of regex captures. Uses the underlying Rure
+    ///     regex iterator.
     /// </summary>
-    public class CapturesIter : RegexIter, IEnumerator<Captures>
+    /// <remarks>Initialise a captures iterator.</remarks>
+    /// <param name="pattern">The pattern to search with.</param>
+    /// <param name="haystack">The haystack to search.</param>
+    public CapturesIter(Regex pattern, byte[] haystack) : base(pattern, haystack)
     {
-        /// <summary>Initialise a captures iterator.</summary>
-        /// <param name="pattern">The pattern to to search with.</param>
-        /// <param name="haystack">The haystack to search.</param>
-        public CapturesIter(Regex pattern, byte[] haystack)
-            : base(pattern, haystack)
-        {}
+    }
 
-        /// <inheritdoc />
-        public Captures Current { get; set; }
 
-        /// <inheritdoc />
-        object IEnumerator.Current => (object)Current;
+    /// <summary>
+    ///     Gets or sets the current capture.
+    /// </summary>
+    public Captures Current { get; set; }
 
-        /// <inheritdoc />
-        public bool MoveNext()
-        {
-            var caps = new Captures(Pattern, Haystack);
-            var matched = RureFfi.rure_iter_next_captures(Raw,
-                                                          Haystack,
-                                                          new UIntPtr((uint)Haystack.Length),
-                                                          caps.Raw);
-
-            if (matched)
-            {
-                caps.Matched = matched;
-                Current = caps;
-                return true;
-            }
+    /// <inheritdoc />
+    object IEnumerator.Current => Current;
 
+    /// <inheritdoc />
+    public bool MoveNext()
+    {
+        if (Pattern == null || Haystack == null || Raw == null)
+        {
             Current = null;
             return false;
         }
 
-        /// <inheritdoc />
-        public void Reset()
+        Captures caps = new(Pattern, Haystack);
+        var matched = RureFfi.rure_iter_next_captures(Raw,
+            Haystack,
+            new UIntPtr((uint)Haystack.Length),
+            caps.Raw);
+
+        if (matched)
         {
-            throw new NotImplementedException();
+            caps.Matched = matched;
+            Current = caps;
+            return true;
         }
+
+        Current = null;
+        return false;
+    }
+
+    /// <inheritdoc />
+    public void Reset()
+    {
+        throw new NotImplementedException();
```

Comment:
II know this isn’t you. But pretty sure this should be not supported rather than not implemented.

Reply:
Regardless, adjusted it.